sourmash_lib/kmer_min_hash.hh

```diff
@@ -190,32 +195,17 @@ public:
 
     std::string _revcomp(const std::string& kmer) const {
         std::string out = kmer;
-        size_t ksize = out.size();
 
-        for (size_t i=0; i < ksize; ++i) {
-            char complement;
+        auto from = out.begin();
+        auto to = out.end();
 
-            switch(kmer[i]) {
-            case 'A':
-                complement = 'T';
-                break;
-            case 'C':
-                complement = 'G';
-                break;
-            case 'G':
-                complement = 'C';
-                break;
-            case 'T':
-                complement = 'A';
-                break;
-            default:
-                std::string msg = "invalid DNA character in sequence: ";
```

Comment:
We don't have any tests that trigger this behavior (and this is different than what `khmer` does)

Reply:
I am confused. We certainly _revcomp a lot....?

Reply:
But `_revcomp` is not exposed to Python, so it was never tested. But I think we can trigger the behavior in `add_hash`, I'm gonna try.

Reply:
No, we can't: `_checkdna` fails before we can use `_revcomp` with an invalid base.